server/sonar-server/src/main/java/org/sonar/server/user/ws/IssueJsonWriter.java

```diff
@@ -0,0 +1,223 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.user.ws;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Multimap;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import org.sonar.api.i18n.I18n;
+import org.sonar.api.issue.ActionPlan;
+import org.sonar.api.issue.Issue;
+import org.sonar.api.issue.IssueComment;
+import org.sonar.api.issue.internal.DefaultIssueComment;
+import org.sonar.api.user.User;
+import org.sonar.api.utils.DateUtils;
+import org.sonar.api.utils.Duration;
+import org.sonar.api.utils.Durations;
+import org.sonar.api.utils.text.JsonWriter;
+import org.sonar.core.component.ComponentDto;
+import org.sonar.markdown.Markdown;
+import org.sonar.server.issue.ws.IssueActionsWriter;
+import org.sonar.server.user.UserSession;
+
+public class IssueJsonWriter {
+
+  private static final String ACTIONS_EXTRA_FIELD = "actions";
+  private static final String TRANSITIONS_EXTRA_FIELD = "transitions";
+  private static final String ASSIGNEE_NAME_EXTRA_FIELD = "assigneeName";
+  private static final String REPORTER_NAME_EXTRA_FIELD = "reporterName";
+  private static final String ACTION_PLAN_NAME_EXTRA_FIELD = "actionPlanName";
+
+  public static final Set<String> EXTRA_FIELDS = ImmutableSet.of(
+    ACTIONS_EXTRA_FIELD, TRANSITIONS_EXTRA_FIELD, ASSIGNEE_NAME_EXTRA_FIELD, REPORTER_NAME_EXTRA_FIELD, ACTION_PLAN_NAME_EXTRA_FIELD);
+
+  private final I18n i18n;
+  private final Durations durations;
+  private final UserSession userSession;
+  private final UserJsonWriter userWriter;
+  private final IssueActionsWriter actionsWriter;
+
+  public IssueJsonWriter(I18n i18n, Durations durations, UserSession userSession, UserJsonWriter userWriter, IssueActionsWriter actionsWriter) {
+    this.i18n = i18n;
+    this.durations = durations;
+    this.userSession = userSession;
+    this.userWriter = userWriter;
+    this.actionsWriter = actionsWriter;
+  }
+
+  public void write(JsonWriter json, Issue issue, Map<String, User> usersByLogin, Map<String, ComponentDto> componentsByUuid,
+    Map<String, ComponentDto> projectsByComponentUuid, Multimap<String, DefaultIssueComment> commentsByIssues, Map<String, ActionPlan> actionPlanByKeys, List<String> extraFields) {
+    json.beginObject();
+
+    String actionPlanKey = issue.actionPlanKey();
+    ComponentDto file = componentsByUuid.get(issue.componentUuid());
+    ComponentDto project = null, subProject = null;
+    if (file != null) {
+      project = projectsByComponentUuid.get(file.uuid());
+      if (!file.projectUuid().equals(file.moduleUuid())) {
+        subProject = componentsByUuid.get(file.moduleUuid());
+      }
+    }
+    Duration debt = issue.debt();
+    Date updateDate = issue.updateDate();
+
+    json
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1067)